go-controller/pkg/cni/cni.go

```diff
@@ -154,6 +160,16 @@ func (pr *PodRequest) cmdAdd(kubeAuth *KubeAPIAuth, clientset *ClientSet) (*Resp
 		if err != nil {
 			return nil, err
 		}
+		if primaryUDN.Found() {
+			primaryUDNPodRequest := pr.buildPrimaryUDNPodRequest(pod, primaryUDN)
+			primaryUDNPodInfo, err := primaryUDNPodRequest.buildPodInterfaceInfo(annotations, primaryUDN.Annotation(), primaryUDN.NetworkDevice())
+			if err != nil {
+				return nil, err
+			}
+			if _, err := primaryUDNPodRequest.getCNIResult(clientset, primaryUDNPodInfo); err != nil {
+				return nil, err
+			}
+		}
 	} else {
 		response.PodIFInfo = podInterfaceInfo
```

Comment:
so it looks like unprivileged mode wont work with UDN right now. Is that right?

you have access to:

type PodInfoGetter interface {
	getPod(namespace, name string) (*kapi.Pod, error)
}

you can make it getNAD or whatever to determine the active network

If you dont want to tackle that now and defer it for later, then at least put a // TODO

Reply:
I think there's nothing technically here preventing us from having it in unprivileged mode.

I am OK with the TODO, then tackle in a follow-up.

Reply:
I will add a TODO